lib/pg_online_schema_change/query.rb

```diff
@@ -319,6 +319,28 @@ def copy_data_statement(client, shadow_table, reuse_trasaction = false)
           FROM ONLY #{client.table}
         SQL
       end
+
+      def primary_key_sequence(shadow_table, primary_key, opened)
+        query = <<~SQL
+          SELECT pg_get_serial_sequence(\'#{shadow_table}\', \'#{primary_key}\') as sequence_name
+        SQL
+
+        result = run(client.connection, query, opened)
+
+        result.map do |row|
+          row["sequence_name"]
+        end&.first
+      end
+
+      def query_for_primary_key_refresh(shadow_table, primary_key, table, opened)
+        sequence_name = primary_key_sequence(shadow_table, primary_key, opened)
+
+        return "" if sequence_name.nil?
+
+        <<~SQL
+          SELECT setval((select pg_get_serial_sequence(\'#{shadow_table}\', \'#{primary_key}\')), (SELECT max(#{primary_key}) FROM #{table})+1);
```

Comment:
In my testing it appears that the `+1` causes a skipped sequence value. I think `setval()` sets the current value (which we get via the `max()` call) which will allow `nextval()` to return the proper next value in the sequence.

https://www.postgresql.org/docs/current/functions-sequence.html

Reply:
ah! nice catch @brycethornton . I have a failing spec now: https://github.com/shayonj/pg-osc/pull/74/files

Should have a fix soon.

Reply:
@jfrost pointed this out here as well: https://github.com/shayonj/pg-osc/pull/72#discussion_r973605106